src/MIDebugEngine/Natvis.Impl/Natvis.cs

```diff
@@ -191,16 +191,44 @@ public void Initialize(string fileName)
             {
                 // failed to find the VS Service
             }
-            string natvisPath = null;
-            string directory = _process.Engine.GetMetric("GlobalVisualizersDirectory") as string;
-            if (!string.IsNullOrEmpty(directory) && !string.IsNullOrEmpty(fileName))
+
+            if (string.IsNullOrEmpty(Path.GetDirectoryName(fileName)))
```

Comment:
Nit: I would revert the order of this if and else statements (so check whether it's not null or empty). I think it makes following the loading logic easier.